src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+    }
+
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals)
+    {
+        deltas[0] = {-delta_vals[1], -delta_vals[1]};
+        deltas[1] = {-delta_vals[1], +delta_vals[1]};
+
+        deltas[2] = {-delta_vals[0], -delta_vals[0]}; // inner square
+        deltas[3] = {-delta_vals[0], +delta_vals[0]};
+        deltas[4] = {+delta_vals[0], -delta_vals[0]};
+        deltas[5] = {+delta_vals[0], +delta_vals[0]};
+
+        deltas[6] = {+delta_vals[1], -delta_vals[1]};
+        deltas[7] = {+delta_vals[1], +delta_vals[1]};
+    }
+};
+
+
+/**************************************************************************/
+template<>
+struct SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Square>
+{
+    template<typename Weights, typename WeightVals>
+    static void set_weights(Weights& weights, WeightVals const& weight_vals)
+    {
+        for (uint8_t i = 0; i < 8; i += 2)
+            weights[i] = weight_vals[1];
+
+        for (uint8_t i = 1; i < 8; i += 2)
+            weights[i] = weight_vals[0];
+    }
+
+    template<typename Deltas, typename DeltaVals>
+    static void set_deltas(Deltas& deltas, DeltaVals const& delta_vals)
+    {
+        deltas[0] = {-delta_vals[1], -delta_vals[1]};
+        deltas[1] = {0.0f, -delta_vals[0]};
+        deltas[2] = {+delta_vals[1], -delta_vals[1]};
+
+        deltas[3] = {-delta_vals[0], 0.0f};
+        deltas[4] = {+delta_vals[0], 0.0f};
+
+        deltas[5] = {-delta_vals[1], delta_vals[1]};
+        deltas[6] = {0.0f, delta_vals[0]};
+        deltas[7] = {+delta_vals[1], delta_vals[1]};
+    }
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<9>>
+    : public ASplitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<9>>
+{
+    using Dispatcher              = ExactSplitParticleDispatcher<DimConst<2>>;
+    static constexpr uint8_t root = 3;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_vals_, deltas, delta_vals_);
+    }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const weight_vals[], //
+                      Deltas& deltas, float const delta_vals[])
+    {
+        weights[4] = weight_vals[ColorIndex::RED];
+
+        for (size_t i : {0, 2, 6, 8}) // corners
+            weights[i] = weight_vals[ColorIndex::BLUE];
+
+        for (size_t i : {1, 3, 5, 7})
+            weights[i] = weight_vals[ColorIndex::GREEN];
+
+        Dispatcher::set_deltas(deltas, delta_vals, root);
+    }
+
+protected:
+    enum ColorIndex : uint8_t { RED = 0, GREEN, BLUE };
+    constexpr static float delta_vals_[]  = {-1, 0, 1};
+    constexpr static float weight_vals_[] = {
+        0.25,   // RED
```

Comment:
should be `BLACK`

Reply:
These colors are based on the PDF https://github.com/PHAREHUB/PHARE/wiki/exactSplit/splitDim2Order1.pdf